### PR TITLE
[ETC] Fixing bugs and refactoring

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Refactoring. Adding checks to the code. 

**Youtrack:** https://youtrack.raccoongang.com/issue/ETC-11

### CODE REVIEW

File: lms/djangoapps/etc_api/views.py [hunk 1]
```diff
@@ -37,36 +38,41 @@ def post(self, request):
...
+        except ValidationError:
             return Response(
                 data={
                     "user_message": "Enter a valid 'username' consisting of letters, numbers, underscores or hyphens."
                 },
                 status=400
             )
 
+        data['name'] = prename
+        data['name'] =  "{} {}".format(data['name'], surname) if data['name'] and surname else data['name']
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: I would use `strip` method:
"{} {}".format(prename, surname).strip() if prename or surname else username

File: lms/djangoapps/etc_api/views.py [hunk 2]
```diff
@@ -37,36 +38,41 @@ def post(self, request):
...
-        except ValidationError as err:
+        except ValidationError:
             return Response(
                 data={
                     "user_message": "Enter a valid 'username' consisting of letters, numbers, underscores or hyphens."
                 },
                 status=400
             )
 
+        data['name'] = prename
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: This line could be omitted

File: lms/djangoapps/etc_api/views.py [hunk 3]
```diff
@@ -37,36 +38,41 @@ def post(self, request):
...
+        data['name'] =  "{} {}".format(data['name'], surname) if data['name'] and surname else data['name']
+        data['name'] = data['name'] if data['name'] else username
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: ditto

File: lms/djangoapps/etc_api/views.py [hunk 4]
```diff
@@ -93,12 +99,16 @@ def post(self, request):
...
         data = request.data
+        user_id = data.get('user_id')
+        if not user_id:
+            return Response(data={"user_message": "'user_id' is required parameter."}, status=400)
+
         try:
-            user = User.objects.get(id=data['user_id'])
-            user.is_active = string_to_boolean(data['is_active'])
+            user = User.objects.get(id=user_id)
+            user.is_active = string_to_boolean(data.get('is_active'))
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: is_active expects boolean please add the check

**Comment by AndreyLykhoman** [comment]
@idegtiarov 
I rewrote code that related with creating 'name' field. I added default value as ''  to 'prename' and 'surname' fields. This is necessary for a new construction. It helps to dismiss 'None' string from the 'name' field.

File: lms/djangoapps/etc_api/views.py [hunk 5]
```diff
@@ -116,17 +125,30 @@ def post(self, request):
...
+                course_id=course_id,
+                defaults={
+                    'mode': data.get('mode', 'honor'),
+                    'is_active': string_to_boolean(data.get('is_active'))
+                }
+            )
+        except InvalidKeyError:
+            return Response(data={"user_message": "Wrong 'course_id'. Course does not exist"}, status=400)
+
+        return Response(data={'enrollment_id': enrollment_obj.id}, status=status.HTTP_200_OK)
```
Comment (idegtiarov) [inline lms/djangoapps/etc_api/views.py]: If you use `HTTP_200_OK` why you don't use `HTTP_400_BAD_REQUEST` in lines L152 and 139

Comment (AndreyLykhoman) [reply lms/djangoapps/etc_api/views.py]: I fixed that mistakes.

**Pull request merged**